sagemaker/pipeline/create_sm_pipeline.py

```diff
@@ -0,0 +1,693 @@
+            A job type from the task_config.jobs_to_run list
+
+        Returns
+        -------
+        Union[ProcessingStep, TrainingStep, None]
+            A pipeline step for the provided job type, or None if the job type is not supported
+        """
+        step_generators = {
+            "gconstruct": self._create_gconstruct_step,
+            "gsprocessing": self._create_gsprocessing_step,
+            "dist_part": self._create_dist_part_step,
+            "gb_convert": self._create_gb_convert_step,
+            "train": self._create_train_step,
+            "inference": self._create_inference_step,
+        }
+        step_generator = step_generators.get(job_type)
+        return step_generator(args) if step_generator else None
+
+    def _create_gconstruct_step(self, args: PipelineArgs) -> ProcessingStep:
+        # Implementation for GConstruct step
+        assert args.graph_construction_config.config_filename, (
+            "Graph construction config file must be specified for GConstruct step. "
+            "Use --graph-construction-config-filename"
+        )
+
+        gconstruct_processor = ScriptProcessor(
+            image_uri=args.aws_config.graphstorm_pytorch_cpu_image_url,
+            role=args.aws_config.execution_role,
+            instance_count=1,
+            instance_type=self.graphconstruct_instance_type_param,
+            command=["python3"],
+            sagemaker_session=self.pipeline_session,
+            volume_size_in_gb=self.volume_size_gb_param,
+        )
+
+        gconstruct_s3_output = Join(
+            on="/",
+            values=[
+                self.output_subpath,
+                "gconstruct",
+            ],
+        )
+
+        gc_local_input_path = "/opt/ml/processing/input"
+        # GConstruct should always be the first step and start with the source data
+        gc_proc_input = ProcessingInput(
+            source=self.input_data_param, destination=gc_local_input_path
+        )
+        gc_local_output_path = "/opt/ml/processing/output"
+        gc_proc_output = ProcessingOutput(
+            source=gc_local_output_path,
+            destination=gconstruct_s3_output,
+            output_name=f"{self.args.task_config.graph_name}-gconstruct",
+        )
+
+        gconstruct_arguments = [
+            "--graph-config-path",
+            self.graphconstruct_config_param,
+            "--input-path",
+            gc_local_input_path,
+            "--output-path",
+            gc_local_output_path,
+            "--graph-name",
+            self.graph_name_param,
+            "--num-parts",
+            self.instance_count_param.to_string(),
+            "--part-method",
+            self.partition_algorithm_param,
+        ]
+
+        # TODO: Make this a pipeline parameter?
+        if args.graph_construction_config.graph_construction_args:
+            gconstruct_arguments.extend(
+                args.graph_construction_config.graph_construction_args.split(" ")
```

Comment:
Have you checked if there is nargs='+' in the gconstruct arg list?

Reply:
Good point, I'll see if we have to adjust the way we parse these args, maybe use the method we use in https://github.com/awslabs/graphstorm/blob/391ff1636ebed735bed17ba57ee7d2f8b894f52b/sagemaker/launch/common_parser.py#L70-L91

Reply:
Only `--output-format` seems to be an nargs='+' argument in GConstruct, so I'd say it's safe to keep this arg as is for this release, and decide whether we want to support multiple output formats for pipelines.

Reply:
Please add a TODO here.